### ISSUE TITLE
Polybar - segmentation fault (core dumped)

### ISSUE THREAD

**Issue by gaabster** [jaagr/polybar#979]
Hi.

So this morning my polybar stopped working after pc restart.
What I was able to find it was a mpd issue.
This is the output of --log=trace:
```
 - config: Copying missing params (sub="module/backlight-acpi", base="module/xbacklight")
* Loaded /home/gabriel/.config/polybar/config
- config: Current bar section: [bar/example]
warn: No monitor specified, using "HDMI-1"
* Loaded monitor HDMI-1 (1920x1080+0+0)
* Bar geometry: 1920x35+0+0
- bar: Attach X event sink
- bar: Attach signal receiver
- controller: Install signal handler
- controller: Setup user-defined modules
- module/bspwm: Deconstructing
error: Disabling module "bspwm" (reason: Could not find socket: /tmp/bspwm_0_0-socket)
- mpdconnection.connect: 127.0.0.1, 6600, "", timeout: 15
error: module/mpd: Connection refused
error: module/xbacklight: Could not get data (err: XCB_NAME (15))
- module/xbacklight: Deconstructing
error: Disabling module "xbacklight" (reason: Not supported for "HDMI-1")
error: Disabling module "volume" (reason: Missing parameter "module/volume.type")
- module/battery: Attach inotify at /sys/class/power_supply/BAT0/energy_now
- module/battery: Attach inotify at /sys/class/power_supply/AC/online
- module/powermenu: Creating menu level 0
- module/powermenu: Creating menu level item 0
- module/powermenu: Creating menu level item 1
- module/powermenu: Creating menu level 1
- module/powermenu: Creating menu level item 0
- module/powermenu: Creating menu level item 1
- module/powermenu: Creating menu level 2
- module/powermenu: Creating menu level item 0
- module/powermenu: Creating menu level item 1
* Starting application
- controller: Main thread id = 1
* Starting module/i3
* Starting module/mpd
* Starting module/filesystem
* Starting module/xkeyboard
- module/mpd: Thread id = 2
* Starting module/memory
[1]    3136 segmentation fault (core dumped)  polybar example --log=trace
```
And this is the output of gdb -ex run -args polybar example:

```
 This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from polybar...done.
Starting program: /usr/local/bin/polybar example
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[New Thread 0x7fffeda24700 (LWP 3732)]
warn: No monitor specified, using "HDMI-1"
error: Disabling module "bspwm" (reason: Could not find socket: /tmp/bspwm_0_0-socket)
error: module/mpd: Connection refused
error: module/xbacklight: Could not get data (err: XCB_NAME (15))
error: Disabling module "xbacklight" (reason: Not supported for "HDMI-1")
error: Disabling module "volume" (reason: Missing parameter "module/volume.type")
[New Thread 0x7fffed223700 (LWP 3735)]
[New Thread 0x7fffeca22700 (LWP 3736)]
[New Thread 0x7fffe7fff700 (LWP 3738)]

Thread 4 "polybar" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffeca22700 (LWP 3736)]
0x00005555558f3af1 in polybar::v3_1_0_35_g5ca1940::mpd::mpdstatus::match_state (this=0x0, state=polybar::v3_1_0_35_g5ca1940::mpd::mpdstate::PLAYING)
    at /home/gabriel/polybar/src/adapters/mpd.cpp:419
419	    return state == m_state;
```

So I commented out the 
`type = internal/mpd`
in the config and it works again.

Now I have to explain few things. My mpd is installed but I didn't configure it yet so no daemon is running, could that be the issue here? I'm running a fresh arch install with i3wm.
Can anyone tell me why this happened?
Thank you and have a good day everyone.

**Comment by gaabster** [comment]
Hi again.

MPD has to be running for polybar to work. 
I don't know if that is how it shloud be. Maybe polybar should just disable the module?
Anyway I'm no expert and don't know how things like that work so I'll be going now.

**Comment by patrick96** [comment]
This is not intended. I was able to reproduce this, it seems this is a bug I've have introduced when I fixed an issue with the mpd time token in 645a3142a19747c252d24cc71414d40a8eb5f7dc

I will try to fix this as soon as possible.

For future reference, here is the full stack trace from someone who also had the same issue.
```
0x08127858 in polybar::v3_1_0_36_g5c7ff09::mpd::mpdstatus::match_state (this=0x0, state=polybar::v3_1_0_36_g5c7ff09::mpd::mpdstate::PLAYING) at /home/evoke/Desktop/polybar/src/adapters/mpd.cpp:419
419	    return state == m_state;
#0  0x08127858 in polybar::v3_1_0_36_g5c7ff09::mpd::mpdstatus::match_state (this=0x0, state=polybar::v3_1_0_36_g5c7ff09::mpd::mpdstate::PLAYING) at /home/evoke/Desktop/polybar/src/adapters/mpd.cpp:419
#1  0x0814c132 in polybar::v3_1_0_36_g5c7ff09::modules::mpd_module::update (this=0x832ae38) at /home/evoke/Desktop/polybar/src/modules/mpd.cpp:203
#2  0x08155f01 in polybar::v3_1_0_36_g5c7ff09::modules::event_module<polybar::v3_1_0_36_g5c7ff09::modules::mpd_module>::runner (this=0x832ae38) at /home/evoke/Desktop/polybar/include/modules/meta/event_module.hpp:23
#3  0xb7a2662e in ?? () from /usr/lib/i386-linux-gnu/libstdc++.so.6
#4  0xb7900295 in start_thread (arg=0xb5c84b40) at pthread_create.c:333
#5  0xb782b0ae in clone () at ../sysdeps/unix/sysv/linux/i386/clone.S:114
```
